src/gui/editors/PianoRoll.cpp

```diff
@@ -778,6 +778,20 @@ void PianoRoll::constrainNoteLengths(bool constrainMax)
 	Engine::getSong()->setModified();
 }
 
+void PianoRoll::flipNotes()
+{
+	if (!hasValidMidiClip()) { return; }
+
+	const NoteVector selectedNotes = getSelectedNotes();
+	const auto& notes = selectedNotes.empty() ? m_midiClip->notes() : selectedNotes;
+
+	m_midiClip->flipNotes(notes);
```

Comment:
I understand that having `Clip::flipNotes` is neat, but I don't like that you can pass notes from any clip to it. I'd rather see that the selection happens inside `Clip::flipNotes`. It's not a deal breaker though... I need a second opinion...